Fixed autoscale in PDAL (division by 0).
--- a/src/pdal/Scaling.cpp
+++ b/src/pdal/Scaling.cpp
@@ -93,26 +93,33 @@
     }
 
     if (m_xXform.m_offset.m_auto)
-    {
         m_xXform.m_offset.m_val = xmin;
-        xmax -= xmin;
-    }
     if (m_yXform.m_offset.m_auto)
-    {
         m_yXform.m_offset.m_val = ymin;
-        ymax -= ymin;
-    }
     if (m_zXform.m_offset.m_auto)
-    {
         m_zXform.m_offset.m_val = zmin;
-        zmax -= zmin;
-    }
+
     if (m_xXform.m_scale.m_auto)
-        m_xXform.m_scale.m_val = xmax / (std::numeric_limits<int>::max)();
+    {
+        xmax = (std::max)(std::abs(xmax - m_xXform.m_offset.m_val),
+                          std::abs(xmin - m_xXform.m_offset.m_val));
+        m_xXform.m_scale.m_val
+            = xmax ? xmax / (std::numeric_limits<int>::max)() : 1.;
+    }
     if (m_yXform.m_scale.m_auto)
-        m_yXform.m_scale.m_val = ymax / (std::numeric_limits<int>::max)();
+    {
+        ymax = (std::max)(std::abs(ymax - m_yXform.m_offset.m_val),
+                          std::abs(ymin - m_yXform.m_offset.m_val));
+        m_yXform.m_scale.m_val
+            = ymax ? ymax / (std::numeric_limits<int>::max)() : 1.;
+    }
     if (m_zXform.m_scale.m_auto)
-        m_zXform.m_scale.m_val = zmax / (std::numeric_limits<int>::max)();
+    {
+        zmax = (std::max)(std::abs(zmax - m_zXform.m_offset.m_val),
+                          std::abs(zmin - m_zXform.m_offset.m_val));
+        m_zXform.m_scale.m_val
+            = zmax ? zmax / (std::numeric_limits<int>::max)() : 1.;
+    }
 }
 
 
