Patch to setup.py for build with Python3. 
--- a/python-support/setup.py
+++ b/python-support/setup.py
@@ -151,13 +151,13 @@
 if platform.system() == 'Darwin':
     extra_link_args.append('-Wl,-rpath,'+library_dirs[0])
 
-DEBUG=True
+DEBUG=False
 if DEBUG:
     if os.name != 'nt':
         extra_compile_args += ['-g','-O0']
 
 # readers.numpy doesn't exist until PDAL 1.8
-if PDALVERSION >= Version('1.8'):
+if "clean" not in sys.argv and PDALVERSION >= Version('1.8'):
     libraries.append('pdal_plugin_reader_numpy')
 
 if os.name in ['nt']:
@@ -197,7 +197,7 @@
 setup_args = dict(
     name                = 'PDAL',
     version             = str(module_version),
-    requires            = ['Python (>=2.7)', 'Numpy'],
+    requires            = ['Python (>=3.0)', 'Numpy'],
     description         = 'Point cloud data processing',
     license             = 'BSD',
     keywords            = 'point cloud spatial',
@@ -217,7 +217,6 @@
         'Intended Audience :: Science/Research',
         'License :: OSI Approved :: BSD License',
         'Operating System :: OS Independent',
-        'Programming Language :: Python :: 2.7',
         'Programming Language :: Python :: 3',
         'Topic :: Scientific/Engineering :: GIS',
     ],
